MovieLibrary/src/com/company/FileUtils.java

```diff
@@ -0,0 +1,88 @@
+package com.company;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileUtils {
+
+
+    private static final String COMMA_DELIMITER = ",";
+
+    public static <T> List<T> readFromFileStream(String fileName) {
+        List<T> result = new ArrayList<>();
+        try {
+            FileInputStream fileInputStream = new FileInputStream(fileName);
+            ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream);
+            while (fileInputStream.available() > 0) {
+                T readObject = (T) objectInputStream.readObject();
+                result.add(readObject);
+            }
+            fileInputStream.close();
+        } catch (IOException | ClassNotFoundException e) {
+            e.printStackTrace();
+            System.out.println(e.getMessage());
+        }
+        return result;
+    }
+    public static boolean fileExists(String fileName) {
+        File file = new File(fileName);
+        return file.exists();
+    }
+
+    public static boolean createFile(String fileName) {
+        File file = new File(fileName);
+        try {
+            return file.createNewFile();
+        } catch (IOException e) {
+            System.out.println("cannot create file: " + fileName);
+        }
+        return false;
+    }
+
+    public static void csvWriter(List<Movie> movies) throws IOException {
+        BufferedWriter out = new BufferedWriter(new FileWriter("csvMovieList"), 512);
+        out.write("id,title,date" + System.lineSeparator());
+        for (Movie movie : movies)
+            out.write(movie.getId() + "," + movie.getTitle() + "," + movie.getDate() + System.lineSeparator());
+        out.close();
+    }
+
+    public static void csvReader(List<Movie> movies) throws IOException, MyException {
+
+        if (!fileExists("csvMovieList)")) {
+            createFile("csvMovieList");
+        }
+        BufferedReader br = new BufferedReader(new FileReader("csvMovieList"));
+        movies.clear();
+        String line;
+        br.readLine();
+        while ((line = br.readLine()) != null) {
+            String[] movieDetails = line.split(COMMA_DELIMITER);
+
+            if (movieDetails.length > 0) {
+                Movie movie = new Movie(Integer.parseInt(movieDetails[0]),
+                        movieDetails[1], Integer.parseInt(movieDetails[2]));
+                movies.add(movie);

```

Comment:
metoda powinna tworzyc nową listę movies, budować ją i zwracac - zamiast modyfikowac tę z parametru wejsciowego.

Jeżeli będziesz chcial dokleic dodatkowe filmy do listy to to nie jest odpowiedzialnosc tej metody - niech to sie dzieje poza ta metoda w konkretnym kodzie z niej korzystajacym